engine/access/rpc/backend/backend.go

```diff
@@ -91,8 +93,17 @@ type Config struct {
 	CircuitBreakerConfig      connection.CircuitBreakerConfig // the configuration for circuit breaker
 }
 
-func New(
-	state protocol.State,
+type Communicator interface {
```

Comment:
please move this to `node_communicator.go`, and add a compiler interface check there as well.
e.g. 
```golang
var _ Communicator = (*NodeCommunicator)(nil)
```

Reply:
appreciate the tip, learned new thing.

Reply:
moved the interface declaration to node_communicator file https://github.com/onflow/flow-go/pull/4598/commits/a6e33aa58afc77f550be9de8fcba4a46af095566